packages/navie/src/user-context.ts

```diff
@@ -0,0 +1,69 @@
+export namespace UserContext {
+  export type CodeSelectionItem = {
+    type: 'code-selection';
+    location: string;
+    content: string;
+  };
+
+  export type CodeSnippetItem = {
+    type: 'code-snippet';
```

Comment:
These do not have an associated location; which actually might not be correct. If the user is Pinning a generated snippet, might that snippet have an associated location?